src/cryptonote_core/miner.cpp

```diff
@@ -411,5 +464,293 @@ namespace cryptonote
+  * Note: add documentation
+  */
+  bool miner::set_is_background_mining_enabled(bool is_background_mining_enabled)
+  {
+    m_is_background_mining_enabled = is_background_mining_enabled;
+    m_is_background_mining_enabled_cond.notify_one();
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  uint64_t miner::get_min_idle_seconds() const
+  {
+    return m_min_idle_seconds;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_min_idle_seconds(uint64_t min_idle_seconds)
+  {
+    if(min_idle_seconds > BACKGROUND_MINING_MAX_MIN_IDLE_INTERVAL_IN_SECONDS) return false;
+    if(min_idle_seconds < BACKGROUND_MINING_MIN_MIN_IDLE_INTERVAL_IN_SECONDS) return false;
+    m_min_idle_seconds = min_idle_seconds;
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  uint8_t miner::get_idle_threshold() const
+  {
+    return m_idle_threshold;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_idle_threshold(uint8_t idle_threshold)
+  {
+    if(idle_threshold > BACKGROUND_MINING_MAX_IDLE_THRESHOLD_PERCENTAGE) return false;
+    if(idle_threshold < BACKGROUND_MINING_MIN_IDLE_THRESHOLD_PERCENTAGE) return false;
+    m_idle_threshold = idle_threshold;
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  uint8_t miner::get_mining_target() const
+  {
+    return m_mining_target;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::set_mining_target(uint8_t mining_target)
+  {
+    if(mining_target > BACKGROUND_MINING_MAX_MINING_TARGET_PERCENTAGE) return false;
+    if(mining_target < BACKGROUND_MINING_MIN_MINING_TARGET_PERCENTAGE) return false;
+    m_mining_target = mining_target;
+    return true;
+  }
+  //-----------------------------------------------------------------------------------------------------
+  bool miner::background_worker_thread()
+  {
+    uint64_t prev_total_time, current_total_time;
+    uint64_t prev_idle_time, current_idle_time;
+    uint64_t previous_process_time = 0, current_process_time = 0;
 
+    if(!get_system_times(prev_total_time, prev_idle_time))
+    {
+      LOG_ERROR(__func__ << " : get_system_times call failed, background mining will NOT work!");
+      return false;
+    }
+    
+    while(!m_stop)
+    {
+        
+      try
+      {
+        // All of this might be overkill if we just enforced some simple requirements
+        // about changing this variable before/after the miner starts, but I envision
+        // in the future a checkbox that you can tick on/off for background mining after
```

Comment:
That would be my recommendation. Because if the command line did not specify background mining, this thread never starts (I think). If that's the case, then the setter does not work as exactly intended in some situations.